### PR TITLE
refactor plugin to use release tooling

### PR DESCRIPTION
**Pull request by kschoche**
moves `main.go` to `cmd/` so the release tooling can build, tag, and release automatically.

### CODE REVIEW

**Review by swenson — APPROVED** [review]
LGTM so far

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ default: dev
 # is only put into /bin/
 .PHONY: quickdev
 quickdev: generate
-	@CGO_ENABLED=0 go build -i -tags='$(BUILD_TAGS)' -o bin/vault-gcp-auth-plugin
+	@CGO_ENABLED=0 go build -i -tags='$(BUILD_TAGS)' -o bin/vault-plugin-auth-gcp cmd/vault-plugin-auth-gcp/main.go
```
Comment (benashz) [inline Makefile:25]: Could use `$(TOOL)` here, instead of repeating it?

**Review by benashz — APPROVED** [review]
One minor nit to take a look at. Otherwise 👍

File: Makefile (hunk 1 above)
Comment (kschoche) [reply Makefile]: Good call, also apparently `-i` flag is invalid now:
```
demo $ make quickdev
flag provided but not defined: -i
usage: go build [-o output] [build flags] [packages]
```
So I removed that as well.
Fixed in a66468f

**Pull request merged**